common/txmgr/strategies.go

```diff
@@ -56,14 +56,15 @@ func (s DropOldestStrategy) Subject() uuid.NullUUID {
 	return uuid.NullUUID{UUID: s.subject, Valid: true}
 }
 
-func (s DropOldestStrategy) PruneQueue(ctx context.Context, pruneService txmgrtypes.UnstartedTxQueuePruner) (n int64, err error) {
+func (s DropOldestStrategy) PruneQueue(ctx context.Context, pruneService txmgrtypes.UnstartedTxQueuePruner) (ids []int64, err error) {
 	var cancel context.CancelFunc
 	ctx, cancel = context.WithTimeout(ctx, s.queryTimeout)
 	defer cancel()
 
-	n, err = pruneService.PruneUnstartedTxQueue(ctx, s.queueSize, s.subject)
+	// NOTE: We prune one less than the queue size because we will be adding a new transaction to the queue right after this PruneQueue call
```

Comment:
This comments helps! But I'm not yet sure why we can't just use the queueSize. The prune is occurring before the create call right? This means that the new transaction isn't in the queue yet, so it should be safe to use s.queueSize as opposed to queueSize - 1 right?